api/src/main/java/org/apache/gravitino/model/ModelCatalog.java

```diff
@@ -0,0 +1,118 @@
+import org.apache.gravitino.exceptions.NoSuchModelException;
+import org.apache.gravitino.exceptions.NoSuchSchemaException;
+
+/**
+ * The ModelCatalog interface defines the public API for managing model objects in a schema. If the
+ * catalog implementation supports model objects, it should implement this interface.
+ */
+@Evolving
+public interface ModelCatalog {
+
+  /**
+   * List the models in a schema namespace from the catalog.
+   *
+   * @param namespace A schema namespace.
+   * @return An array of model identifiers in the namespace.
+   * @throws NoSuchSchemaException If the schema does not exist.
+   */
+  NameIdentifier[] listModels(Namespace namespace) throws NoSuchSchemaException;
+
+  /**
+   * Get a model metadata by {@link NameIdentifier} from the catalog.
+   *
+   * @param ident A model identifier.
+   * @return The model metadata.
+   * @throws NoSuchModelException If the model does not exist.
+   */
+  Model getModel(NameIdentifier ident) throws NoSuchModelException;
+
+  /**
+   * Check if a model exists using an {@link NameIdentifier} from the catalog.
+   *
+   * @param ident A model identifier.
+   * @return true If the model exists, false if the model does not exist.
+   */
+  default boolean modelExists(NameIdentifier ident) {
+    try {
+      getModel(ident);
+      return true;
+    } catch (NoSuchModelException e) {
+      return false;
+    }
+  }
+
+  /**
+   * Register a model in the catalog if the model is not existed, otherwise the {@link
+   * ModelAlreadyExistsException} will be thrown. The {@link Model} object will be created the model
+   * is registered, users can call {@link Model#link(String[], String, String, Map)} to link the
+   * model version to the registered {@link Model}.
+   *
+   * @param ident The name identifier of the model.
+   * @param comment The comment of the model. The comment is optional and can be null.
+   * @param properties The properties of the model. The properties are optional and can be null or
+   *     empty.
+   * @return The registered model object.
+   * @throws ModelAlreadyExistsException If the model already registered.
+   */
+  default Model registerModel(NameIdentifier ident, String comment, Map<String, String> properties)
+      throws ModelAlreadyExistsException {
+    return registerModel(ident, new String[0], comment, null, properties);
+  }
+
+  /**
+   * Register a model in the catalog if the model is not existed, otherwise the {@link
+   * ModelAlreadyExistsException} will be thrown. The {@link Model} object will be created when the
+   * model is registered, in the meantime, the model version (version 0) will also be created and
+   * linked to the registered model.
+   *
+   * @param ident The name identifier of the model.
+   * @param aliases The aliases of the model version. The alias are optional and can be empty.
+   * @param comment The comment of the model. The comment is optional and can be null.
+   * @param url The model artifact URL.
```

Comment:
What is the difference between `url` here and `uri` in `ModelVersion`? why do they have different names

Reply:
It should be uri, I should have changed all the places, but this might be the left one.